openraft/tests/fixtures/mod.rs

```diff
@@ -913,11 +927,17 @@ where
     S: Default + Clone,
 {
     type Network = RaftRouterNetwork<C, S>;
+    type ConnectError = NetworkError;
 
-    async fn connect(&mut self, target: C::NodeId, _node: Option<&Node>) -> Self::Network {
-        RaftRouterNetwork {
-            target,
-            owner: self.clone(),
+    async fn connect(&mut self, target: C::NodeId, _node: Option<&Node>) -> Result<Self::Network, NetworkError> {
+        let e = NetworkError::new(&AnyError::error(format!("failed to connect: {}", target)));
+        if self.network_failed.load(Ordering::Relaxed) {
+            Err(e)
```

Comment:
`e` does not need to be in the outer block: build such an `e` only when `network_failed` is true.